src/zcash/address/unified.cpp

```diff
@@ -61,10 +71,12 @@ ZcashdUnifiedFullViewingKey ZcashdUnifiedFullViewingKey::FromUnifiedFullViewingK
     return result;
 }
 
-std::optional<UnifiedAddress> ZcashdUnifiedFullViewingKey::Address(diversifier_index_t j) const {
+std::optional<UnifiedAddress> ZcashdUnifiedFullViewingKey::Address(
+        const diversifier_index_t& j,
+        const std::set<ReceiverType>& receiverTypes) const
+{
     UnifiedAddress ua;
-
-    if (saplingKey.has_value()) {
+    if (saplingKey.has_value() && receiverTypes.count(ReceiverType::Sapling) > 0) {
```

Comment:
In the PR that connects this logic to the `z_getaddressforaccount` RPC method, the latter's help text needs to document that "unsupported" receiver types for this account (e.g. if the account was generated before some future shielded protocol was added to `zcashd`) will be ignored, i.e. the returned UA will contain the largest subset of the specified receiver types that it can support. (This behaviour might change in future, if e.g. we decide to automatically add future shielded pools to existing accounts, but doing so is compatible with this documentation.)